contracts/modules/staking/PooledStaking.sol

```diff
@@ -737,47 +957,519 @@ contract PooledStaking is IPooledStaking, MasterAware {
+      return 3642751750084770400; // 3.6427517500847704%
+    }
+
+    // Idle DAI v4 (idleDAIYield)
+    if (id == 101) {
+      return 4831821144747536000; // 4.831821144747536%
+    }
+
+    // Idle USDT v4 (idleUSDTYield)
+    if (id == 102) {
+      return 5089820795846740000; // 5.08982079584674%
+    }
+
+    // Pool Together v4
+    if (id == 110) {
+      return 13592770586860942000; // 13.592770586860942%
+    }
+
+    // Trader Joe
+    if (id == 111) {
+      return 7995085516794736000; // 7.9950855167947354%
+    }
+    // {V1_PRICES_HELPER_END}
+
+    revert("Invalid product id");
+  }
+
+  function migrateToNewV2Pool(address stakerAddress, uint groupId) external noPendingActions {
+    require(block.timestamp <= migrationDeadline, "Migration period has ended");
     // Addresses marked for implicit migration can be migrated by anyone.
     // Addresses who are not can only be migrated by calling this function themselves.
-    // [todo] Add the marked addresses before deploy
+    // [todo] Check these addresses before deploy
     require(
       stakerAddress == msg.sender ||
-      stakerAddress == 0x0000000000000000000000000000000000000001 ||
-      stakerAddress == 0x0000000000000000000000000000000000000002, // etc.
+      stakerAddress == 0x1337DEF1FC06783D4b03CB8C1Bf3EBf7D0593FC4 || // Armor 48%
+      stakerAddress == 0x87B2a7559d85f4653f13E6546A14189cd5455d45 || // Hugh 16.3%
+      stakerAddress == 0x4a9fA34da6d2378c8f3B9F6b83532B169beaEDFc || // 6.6%
+      stakerAddress == 0x46de0C6F149BE3885f28e54bb4d302Cb2C505bC2 || // 4.5%
+      stakerAddress == 0xE1Ad30971b83c17E2A24c0334CB45f808AbEBc87 || // 2.5%
+      stakerAddress == 0x5FAdEA9d64FFbe0b8A6799B8f0c72250F92E2B1d || // 1.7%
+      stakerAddress == 0x9c657DB2B697846BE13Ca0B2bB5a6D17f860a395 || // 1.5%
+      stakerAddress == 0xF99b3a13d46A04735BF3828eB3030cfED5Ea0087 || // 1.4%
+      stakerAddress == 0x8C878B8f805472C0b70eD66a71c0B33da3d233c8 || // 1.4%
+      stakerAddress == 0x4544e2Fae244eA4Ca20d075bb760561Ce5990DC3, // 0.7%
       "You are not authorized to migrate this staker"
     );
 
+    require(stakers[stakerAddress].deposit > 0, "Address has no migratable stake");
+    console.log("Migrating staker %s", stakerAddress);
 
     uint contractsCount = stakers[stakerAddress].contracts.length;
     uint deposit = stakers[stakerAddress].deposit;
 
-    uint[] memory weights = new uint[](contractsCount);
     uint[] memory products = new uint[](contractsCount);
+    uint[] memory stakes = new uint[](contractsCount);
+    uint migratableCount = 0;
+
     for (uint i = 0; i < contractsCount; i++) {
       address oldProductId = stakers[stakerAddress].contracts[i];
-      uint productId = productsV1.getNewProductId(oldProductId);
+      uint productId;
+      try productsV1.getNewProductId(oldProductId) returns (uint v) {
+        productId = v;
+      } catch {
+        console.log("oldProductId %s not found in productsV1, likely deprecated", oldProductId);
```

Comment:
now that i see this, should we leave the console.log in the deploy to have this logging?

Or just have a custom event, which adds less to the bytecode